include/Global.h

```diff
@@ -30,7 +30,7 @@ extern double     dTime_Base;                         // physical time interval
 extern double     FlagTable_Rho        [NLEVEL-1];    // refinement criterion of density
 extern double     FlagTable_RhoGradient[NLEVEL-1];    // refinement criterion of density gradient
 extern double     FlagTable_Lohner     [NLEVEL-1][4]; // refinement criterion based on Lohner's error estimator
-extern double     FlagTable_User       [NLEVEL-1];    // user-defined refinement criterion
+extern double    *FlagTable_User       [NLEVEL-1];    // user-defined refinement criterion
```

Comment:
Would it be easier to declare this array as `(*FlagTable_User)[NLEVEL-1]` instead of `*FlagTable_User[NLEVEL-1]`? Given that the number of variables on each level is always `OPT__FLAG_USER_NUM`, we don't really need a pointer array here.

In that case, this array can be allocated by `FlagTable_User = new double [OPT__FLAG_USER_NUM][NLEVEL]`. The corresponding HDF5 variable can be manipulated easier by following, for example, [FlagTable_Lohner](https://github.com/gamer-project/gamer/blob/master/include/HDF5_Typedef.h#L614).

 What do you think? Since this may affect several places in this PR, I'll wait for your response before looking into other modifications.

Reply:
I agree. It would take time to update the code to array pointer.

Reply:
IIUC, in this way the allocated `FlagTable_User` is a `OPT__FLAG_USER_NUM` by `NLEVEL - 1` array. That means we need to pass each **column** of `FlagTable_User` in the `Flag_User_Ptr` function. The common way is to pass the pointer to the first element in each column, and then use indices ptr, ptr + (NLEVEL - 1), ptr + 2 * (NLEVEL -1), etc, to obtain the value (instead of ptr, ptr + 1, ...). This approach could be not user-friendly. The possible solutions could be:

1. Declare `FlagTable_User` as a pointer to pointer via `double **FlagTable_User`, and call `Aux_AllocateArray2D` to allocate the array with size `[NLEVEL - 1] [OPT__FLAG_USER_NUM]`
2. Use pointer array as shown in this PR.

What do you think?

Reply:
@hfhsieh Good suggestion. Let's follow your original approach. I'll check it out soon.